llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1443,6 +1443,17 @@ InstructionCost RISCVTTIImpl::getCmpSelInstrCost(unsigned Opcode, Type *ValTy,
     }
   }
 
+  // With ShortForwardBranchOpt or ConditionalMoveFusion, scalar icmp + select
+  // instructions will lower to SELECT_CC and lower to PseudoCCMOVGPR which will
+  // generate a conditional branch + mv. The cost of scalar (icmp + select) will
+  // be (0 + select instr cost).
+  if (ST->hasConditionalMoveFusion() && I && isa<ICmpInst>(I) &&
+      !ValTy->isVectorTy() && I->hasOneUser() &&
+      isa<SelectInst>(I->user_back()) &&
+      !I->user_back()->getType()->isVectorTy() &&
```

Comment:
Should we just check `I->user_back()->getType()->isIntegerTy()` rather than `!I->user_back()->getType()->isVectorTy()`? It can also filter out something like icmp + select with fp value.